docs/content/standards/currency.mdx

```diff
@@ -0,0 +1,477 @@
+**`CoinRegistry`:** The main system object that coordinates all currency-related operations. This shared object serves as the entry point for all registry operations and is created once during network initialization.
+
+**`Currency<T>`:** The core of the registry system, storing comprehensive information about each coin type including:
+- **Metadata management:** Basic currency information like name, symbol, decimals, description, and icon URL.
+- **Supply tracking:** Maintains supply state information (fixed, burn-only, or unknown).
+- **Regulatory status:** Tracks whether the coin is regulated with deny list capabilities.
+- **Capability references:** Links to `TreasuryCap`s and `MetadataCap`s for the coin type.
+- **Extensibility:** Includes extra fields for future enhancements.
+
+#### Supply states
+
+The registry supports three different supply management models:
+
+- **Fixed supply:** The total supply is permanently locked and cannot be changed.
+- **Burn-only supply:** New coins cannot be minted, but existing coins can be burned.
+- **Unknown supply:** Traditional model where the `TreasuryCap` holder controls minting and burning.
+
+#### Regulatory states
+
+Currencies can have different regulatory statuses:
+
+- **Regulated:** The currency has an associated `DenyCapV2` that can restrict addresses from using it.
+- **Unregulated:** The currency was created without any deny list capabilities.
+- **Unknown:** Regulatory status is undetermined (typically from legacy migrations).
+
+#### Creation options
+
+The registry supports two different currency creation flows:
+
+- **Standard creation (recommended):**
+
+    Use the `new_currency` function at any time after the currency type is published. The function immediately creates a shared `Currency<T>` object.
+
+    <ImportContent source="crates/sui-framework/packages/sui-framework/sources/registries/coin_registry.move" mode="code" fun="new_currency" signatureOnly noComments />
+
+- **One-Time Witness (OTW) creation:**
+
+    :::caution 
+
+    OTW creation of a new currency is a two-step process. The initialization process begins with package publication. Then a call to `finalize_registration` is needed to place the currency into the registry.
+
+    :::
+
+    The `new_currency_with_otw` function uses an OTW for uniqueness proof.
+
+    <ImportContent source="crates/sui-framework/packages/sui-framework/sources/registries/coin_registry.move" mode="code" fun="new_currency_with_otw" signatureOnly noComments />
+
+#### Currency initialization
+
+Both creation methods return a `CurrencyInitializer<T>` that allows for additional configuration:
+
+- **Make regulated:** Add deny list capabilities.
+- **Set supply model:** Choose between fixed, burn-only, or flexible supply.
+- **Add extensions:** Include additional fields for custom functionality.
+
+<ImportContent source="crates/sui-framework/packages/sui-framework/sources/registries/coin_registry.move" mode="code" struct="CurrencyInitializer" />
```

Comment:
In rendered docs says "Struct is not found"
<img width="705" height="398" alt="image" src="https://github.com/user-attachments/assets/59a19081-31bc-42e6-932e-bb59ebd5683c" />